test/e2e/specs/editor/various/block-deletion.spec.js

```diff
@@ -0,0 +1,343 @@
+		).toBeFocused();
+
+		// Hit backspace to remove the empty paragraph.
+		await page.keyboard.press( 'Backspace' );
+
+		// Ensure the last block was removed.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: 'First' } },
+			{ name: 'core/paragraph', attributes: { content: 'Second' } },
+		] );
+
+		// Ensure the caret is in a correct position.
+		await page.keyboard.type( '| ← caret was here' );
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: 'First' } },
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Second| ← caret was here' },
+			},
+		] );
+	} );
+
+	test( 'deleting last selected block via backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		// Add a couple of paragraphs.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'First' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Second' },
+		} );
+
+		// Add the Image block and focus it.
+		await editor.insertBlock( {
+			name: 'core/image',
+		} );
+		const imageBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Image',
+		} );
+		await expect(
+			imageBlock.getByRole( 'button', { name: 'Upload' } )
+		).toBeFocused();
+		await page.keyboard.press( 'ArrowUp' );
+		await expect( imageBlock ).toBeFocused();
+
+		// Hit backspace and ensure the Image block was removed.
+		await page.keyboard.press( 'Backspace' );
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: 'First' } },
+			{ name: 'core/paragraph', attributes: { content: 'Second' } },
+		] );
+
+		// Ensure the caret is in a correct position.
+		await page.keyboard.type( '| ← caret was here' );
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: 'First' } },
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Second| ← caret was here' },
+			},
+		] );
+	} );
+
+	test( 'deleting the last two selected blocks via backspace', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Add a couple of paragraphs.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'First' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Second' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Third' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+
+		// Ensure the empty paragraph is focused.
+		await expect(
+			editor.canvas.getByRole( 'document', {
+				name: /Empty block/i,
+			} )
+		).toBeFocused();
+
+		// Select the last two paragraphs.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await expect(
+			editor.canvas.locator( '.is-multi-selected' )
+		).toHaveCount( 2 );
```

Comment:
Nit: I did a similar thing in #48035 using the `getMultiSelectedBlocks` API. Maybe we can do something like that here?

```js
await expect.poll( () =>
	window.wp.data.select('core/block-editor').getMultiSelectedBlocks()
).toHaveCount( 2 );
```

Reply:
I guess we're not testing the multi-block selection here, so it's OK to validate programmatically! 👍

Reply:
Even if we do, I don't think `.is-multi-selected` should be a public API for end users, so it should never be a testing target IMO.

Reply:
Could you elaborate on why you think it should not be a public API?

Reply:
It's a class that's never exposed to the users, nor to the plugin developers. On the other hand, [`getMultiSelectedBlocks`](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#getmultiselectedblocks) is a documented API. While it's still not a user-facing public API, but it's better than nothing I guess. A better way might be to check for accessible attributes, like `aria-selected` or `aria-current`, but that's probably not going to be available. The next option is visual testing, but not more difficult, and should only be tested once.